File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/exception/BaseException.java
package com.zumin.sudoku.common.core.exception;

import com.zumin.sudoku.common.core.enums.StatusCode;
import lombok.Getter;

/**
 * 基础异常类
 */
@Getter
public class BaseException extends RuntimeException {

  private static final long serialVersionUID = -8105492018397966401L;

  /**
   * 异常编号
   */
  private final StatusCode statusCode;

  public BaseException(StatusCode statusCode) {
    this.statusCode = statusCode;
  }

  public BaseException(StatusCode statusCode, Throwable cause) {
    super(cause);
    this.statusCode = statusCode;
  }
}

File: sudoku-common/common-core/src/main/java/com/zumin/sudoku/common/core/exception/BaseException.kt
package com.zumin.sudoku.common.core.exception

import com.zumin.sudoku.common.core.code.StatusCode

/**
 * 基础异常类
 */
open class BaseException : RuntimeException {

  /**
   * 异常编号
   */
  val statusCode: StatusCode

  constructor(statusCode: StatusCode) {
    this.statusCode = statusCode
  }

  constructor(statusCode: StatusCode, cause: Throwable) : super(cause) {
    this.statusCode = statusCode
  }

  companion object {
    private const val serialVersionUID: Long = 5728954500514726698L
  }
}